src/core/stream.h

```diff
@@ -122,6 +122,9 @@ typedef union QUIC_STREAM_FLAGS {
         BOOLEAN LocalNotAllowed         : 1;    // Peer's unidirectional stream.
         BOOLEAN LocalCloseFin           : 1;    // Locally closed (graceful).
         BOOLEAN LocalCloseReset         : 1;    // Locally closed (locally aborted).
+        BOOLEAN LocalCloseResetReliable : 1;    // Indicates that we should shutdown the send path once we sent/ACK'd ReliableOffsetSend bytes.
+        BOOLEAN LocalCloseResetReliableAcked : 1; // Indicates the peer has acknowledged we will stop sending once we sent/ACK'd ReliableOffsetSend bytes.
```

Comment:
Can this flag be replaced by reusing `LocalCloseAcked`? `LocalCloseAcked && LocalCloseAcked` essentially equals this, right?

Reply:
Not sure, but even if we could I would much prefer the separation, because LocalCloseAcked is used in soooo many other different places we could potentially risk regression if we overextend its usage to this preview feature. The flags here are all internally used, no? An app is not concerned with this stuff.

Reply:
Just because a variable is used in a lot of places doesn't mean is usage is overextended. `LocalCloseAcked` is meant to indicate whatever close mechanism has been used has now been acknowledged. This is a fairly clean design because it effectively means the send direction is 100% done now. Having multiple variables that mean effectively the same thing means you complicated the "are we done yet" logic by requiring more variables to be checked.